<?xml version="1.0"?>
<launch>
  <arg name="use_dji" default="True" />
  <arg name="headless" default="False" />
  <arg name="use_mapping" default="True" />
  <arg name="uav_odom_topic_name" default="/modified_odom" />

  ############## Viewer ###############
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find jsk_uav_forest_common)/launch/debug_viewer.rviz" unless="$(arg headless)" respawn="true"/>
  <param name="robot_description" command="$(find xacro)/xacro '$(find jsk_uav_forest_simulation)/urdf/quadrotor_cam_hokuyo.gazebo.xacro' base_link_frame:=base_link world_frame:=world" />

  ######### Perception Node #########
  <include file="$(find jsk_uav_forest_perception)/launch/tree_detector.launch">
    <arg name="simulation" value="True" unless="$(arg use_dji)" />
    <arg name="uav_odom_topic_name" value="$(arg uav_odom_topic_name)" />
  </include>

  ######## 2D SLAM ##################
  <include file="$(find jsk_uav_forest_common)/launch/2d_mapping.launch" if="$(arg use_mapping)" >
  </include >

</launch>
